test/atom_finder/classifier_test.clj

```diff
@@ -172,3 +172,40 @@
         (testing (str "Is reversed-subscript-atom? - " code " - " sc?)
           (is (= (reversed-subscript-atom? (parse-expr code)) sc?))))
       )))
+
+(deftest test-omitted-curly-braces-atom?
+  (testing "omitted-curly-braces-atom? finds all atoms in snippet study code"
+
+    (let [lines  (->> (resource-path "omitted-curly-braces.c")
+                      tu
+                      (filter-tree omitted-curly-braces-atom?)
+                      (map loc)
+                      (map :line))]
+
+      (is (= lines [3 4 7 8 26 27 28 29 41 43 44 54 66 79 95 101 102 104 105 108 109])))))
+
+
+(deftest comma-operator-test
+  (testing "small statements" 
+    (is (comma-operator-atom? (parse-expr "1,2")))
+    (is (false? (comma-operator-atom? (parse-expr "int a,b;"))))
+    (is (comma-operator-atom? (parse-expr "num1,num2")))
+    (let [lines  (->> (resource-path "comma-operator.c")
+                      tu
+                      (filter-tree comma-operator-atom?)
+                      (map loc)
+                      (map :line))]
+
+      (is (= lines [4 11 11 11])))))
+
+(deftest test-assignment-as-value-atom?
+  (testing "assignment-as-value-atom? finds all atoms in snippet study code"
+
+    (let [lines  (->> (resource-path "assignment-as-value.c")
+                      tu
+                      (filter-tree assignment-as-value-atom?)
+                      (map loc)
+                      (map :line))]
+
+      (is (= lines [8 10 12 12 14 16 20 22 30 32 36 38 42 48 50 52 52 56 58 60 62])))))
```

Comment:
Before when I mentioned the \<true\>/\<false\> test thing, I was talking about the `test-atom-lines` macro. You call it like this:

`(test-atom-lines "type-conversion.c" "<true>" (default-finder type-conversion-atom?)))`

What it does is it parses the file in the first argument (in this case `type-conversion.c`), and it finds every line that has a pattern in it (in this case `<true>`), and it makes sure that the atom returns true for each of those lines. It's the same as doing `(is (= lines [8 10 12 ...]))` except that the list of lines is built automatically based on which lines in your C file have `<true>` written on them.

Please swap out this test to use `test-atom-lines`. You can see a full example in `type_conversion_test.clj`.